src/main/java/com/google/devtools/build/lib/buildeventstream/BuildEvent.java

```diff
@@ -53,10 +53,18 @@
 
     public final Path path;
     public final LocalFileType type;
+    public final boolean shouldUpload; /** whether the local file should be uploaded to the BEP */
```

Comment:
afaik this is unused?

Reply:
It is used for the bep_disable_all_artifact_upload flag here: https://github.com/arlyon/bazel/blob/eb94fe6caddfe89bcc396be556160f30a1bb618f/src/main/java/com/google/devtools/build/lib/remote/ByteStreamBuildEventArtifactUploader.java#L103

Reply:
And here is where the flag is detected: https://github.com/arlyon/bazel/blob/0a5961bd0e6594800ad5cd9a26a641472d6466cd/src/main/java/com/google/devtools/build/lib/buildeventservice/BuildEventServiceUploader.java#L178

If the flag is provided, any artifacts referenced by build events are not uploaded. Perhaps a better name would be `shouldUploadArtifacts`